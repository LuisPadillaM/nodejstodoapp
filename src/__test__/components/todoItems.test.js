import React from 'react';
import {shallow} from 'enzyme';
import TodoItems from 'Components/todoItems.js';
import renderer from 'react-test-renderer';

test('Todo Items render test', () => {

  const itemsMock = [
    {
      "id": 23869840,
      "task": "doSomething"
    },
    {
      "id": 23869840,
      "task": "doSomethingElse"
    },
  ]

  let defaultF = (e) => e

  const component = renderer.create(
    <TodoItems items={itemsMock} deleteItem={defaultF} editItem={defaultF} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  //manually trigger deleteItem
  // tree.props.deleteItem();
});
