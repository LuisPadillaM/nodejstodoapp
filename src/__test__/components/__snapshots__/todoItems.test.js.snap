// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Items render test 1`] = `
<ul
  className=""
  data-react-toolbox="list"
>
  <span>
    <li
      className="undefined "
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className=""
      >
        <span
          className=""
        >
          <span
            className="undefined"
            data-react-toolbox="list-item-text"
          >
            doSomething
          </span>
        </span>
        <span
          className={undefined}
        >
          <span
            className={undefined}
            onClick={undefined}
            onMouseDown={undefined}
          >
            <span
              aria-label=""
              className=""
              data-react-toolbox="font-icon"
            >
              <span>
                <span
                  aria-label=""
                  className="material-icons"
                  data-react-toolbox="font-icon"
                  onClick={[Function]}
                >
                  edit
                </span>
                <span
                  aria-label=""
                  className="material-icons"
                  data-react-toolbox="font-icon"
                  onClick={[Function]}
                >
                  delete
                </span>
              </span>
            </span>
          </span>
        </span>
      </span>
    </li>
    <div
      className="hide"
    >
      <div
        className="editInput"
      >
        <div
          className=""
          data-react-toolbox="input"
        >
          <input
            className="undefined"
            defaultValue={undefined}
            disabled={false}
            maxLength={undefined}
            name={undefined}
            onChange={[Function]}
            onKeyPress={undefined}
            required={false}
            role="input"
            type="text"
            value="doSomething"
          />
          <span
            className={undefined}
          />
          <label
            className=""
          >
            Edit
          </label>
        </div>
      </div>
      <div
        className="icons"
      >
        <span>
          <span
            aria-label=""
            className="material-icons"
            data-react-toolbox="font-icon"
            onClick={[Function]}
          >
            done
          </span>
          <span
            aria-label=""
            className="material-icons"
            data-react-toolbox="font-icon"
            onClick={[Function]}
          >
            clear
          </span>
        </span>
      </div>
    </div>
    <hr
      className={undefined}
    />
  </span>
  <span>
    <li
      className="undefined "
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className=""
      >
        <span
          className=""
        >
          <span
            className="undefined"
            data-react-toolbox="list-item-text"
          >
            doSomethingElse
          </span>
        </span>
        <span
          className={undefined}
        >
          <span
            className={undefined}
            onClick={undefined}
            onMouseDown={undefined}
          >
            <span
              aria-label=""
              className=""
              data-react-toolbox="font-icon"
            >
              <span>
                <span
                  aria-label=""
                  className="material-icons"
                  data-react-toolbox="font-icon"
                  onClick={[Function]}
                >
                  edit
                </span>
                <span
                  aria-label=""
                  className="material-icons"
                  data-react-toolbox="font-icon"
                  onClick={[Function]}
                >
                  delete
                </span>
              </span>
            </span>
          </span>
        </span>
      </span>
    </li>
    <div
      className="hide"
    >
      <div
        className="editInput"
      >
        <div
          className=""
          data-react-toolbox="input"
        >
          <input
            className="undefined"
            defaultValue={undefined}
            disabled={false}
            maxLength={undefined}
            name={undefined}
            onChange={[Function]}
            onKeyPress={undefined}
            required={false}
            role="input"
            type="text"
            value="doSomethingElse"
          />
          <span
            className={undefined}
          />
          <label
            className=""
          >
            Edit
          </label>
        </div>
      </div>
      <div
        className="icons"
      >
        <span>
          <span
            aria-label=""
            className="material-icons"
            data-react-toolbox="font-icon"
            onClick={[Function]}
          >
            done
          </span>
          <span
            aria-label=""
            className="material-icons"
            data-react-toolbox="font-icon"
            onClick={[Function]}
          >
            clear
          </span>
        </span>
      </div>
    </div>
    <hr
      className={undefined}
    />
  </span>
</ul>
`;
